src/js/components/stock-movement-wizard/combined-shipments/AddItemsPage.jsx

```diff
@@ -584,17 +602,57 @@ class AddItemsPage extends Component {
    * @public
    */
   saveAndTransitionToNextStep(formValues, lineItems) {
-    if (_.some(lineItems, item => item.inventoryItem
-      && item.expirationDate !== item.inventoryItem.expirationDate)) {
-      if (_.some(lineItems, item => item.inventoryItem && item.inventoryItem.quantity && item.inventoryItem.quantity !== '0')) {
-        this.confirmInventoryItemExpirationDateUpdate(() =>
-          this.saveRequisitionItemsAndTransitionToNextStep(formValues, lineItems));
-      } else {
-        this.saveRequisitionItemsAndTransitionToNextStep(formValues, lineItems);
-      }
-    } else {
-      this.saveRequisitionItemsAndTransitionToNextStep(formValues, lineItems);
+    if (_.some(lineItems, item =>
+      item?.expirationDate &&
+      item?.expirationDate !== item?.fetchedInventoryItem?.inventoryItem?.expirationDate &&
+      item?.fetchedInventoryItem?.quantityOnHand > 0)) {
+      this.confirmInventoryItemExpirationDateUpdate(() =>
+        this.saveRequisitionItemsAndTransitionToNextStep(formValues, lineItems));
+      return;
+    }
+    this.saveRequisitionItemsAndTransitionToNextStep(formValues, lineItems);
+  }
+
+  async fetchInventoryItem(values, rowIndex, isDebounced = false) {
+    if (isDebounced) {
```

Comment:
Why you can't just call `fetchInventoryItem` or `debouncedInventoryItem` when needed? I feel like this nesting one into another will make a headache in the future.